src/publish_control.cpp

```diff
@@ -114,6 +133,481 @@ void PublishControl::callback_veh_speed(const pacmod_msgs::VehicleSpeedRpt::Cons
   speed_mutex.unlock();
 }
 
+void PublishControl::callback_shift_rpt(const pacmod_msgs::SystemRptInt::ConstPtr& msg)
+{
+  shift_mutex.lock();
+  // Store the latest value read from the gear state to be sent on enable/disable
+  last_shift_cmd = msg->output;
+  shift_mutex.unlock();
+}
+
+void PublishControl::callback_turn_rpt(const pacmod_msgs::SystemRptInt::ConstPtr& msg)
+{
+  turn_mutex.lock();
+  // Store the latest value read from the gear state to be sent on enable/disable
+  last_turn_cmd = msg->output;
+  turn_mutex.unlock();
+}
+
+void PublishControl::callback_rear_pass_door_rpt(const pacmod_msgs::SystemRptInt::ConstPtr& msg)
+{
+  rear_pass_door_mutex.lock();
+  // Store the latest value read to be sent on enable/disable
+  last_rear_pass_door_cmd = msg->output;
+  rear_pass_door_mutex.unlock();
+}
+
+// Publishing
+void PublishControl::publish_steering_message(const sensor_msgs::Joy::ConstPtr& msg)
+{
+  // Steering
+  // Axis 0 is left thumbstick, axis 3 is right. Speed in rad/sec.
+  pacmod_msgs::SteerSystemCmd steer_msg;
+
+  steer_msg.enable = local_enable;
+  steer_msg.ignore_overrides = false;
+
+  // If the enable flag just went to true, send an override clear and clear_faults
+  if (!prev_enable && local_enable)
+  {
+    steer_msg.clear_override = true;
+    steer_msg.clear_faults = true;
+  }
+
+  float range_scale;
+  if (vehicle_type == VEHICLE_4 ||
+      vehicle_type == VEHICLE_6 ||
+      vehicle_type == LEXUS_RX_450H ||
+      vehicle_type == FREIGHTLINER_CASCADIA ||
+      vehicle_type == JUPITER_SPIRIT)
+    range_scale = 1.0;
+  else
+    range_scale = fabs(msg->axes[axes[steering_axis]]) * (STEER_OFFSET - ROT_RANGE_SCALER_LB) + ROT_RANGE_SCALER_LB;
+
+  float speed_scale = 1.0;
```

Comment:
rename it to `steer_scale`?

Reply:
Added to a refactor issue: https://github.com/astuff/pacmod_game_control/issues/85